File: web/src/api/accuracy.ts
import type { JobResponse, RunValidationRequest, RunAllRequest, RunIncidentsRequest, DiscoveryRequest, DiscoveryResponse } from "@/types";

const BASE_URL = import.meta.env.VITE_API_BASE_URL ?? "";

async function handleResponse<T>(res: Response): Promise<T> {
  if (!res.ok) {
    throw new Error(`Request failed with status ${res.status}`);
  }
  return res.json() as Promise<T>;
}

export async function listAccuracyScripts(): Promise<string[]> {
  const res = await fetch(`${BASE_URL}/api/accuracy/scripts`);
  return handleResponse<string[]>(res);
}

export async function runValidation(req: RunValidationRequest): Promise<JobResponse> {
  const res = await fetch(`${BASE_URL}/api/accuracy/run`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(req),
  });
  return handleResponse<JobResponse>(res);
}

export async function runAllValidations(req: RunAllRequest): Promise<JobResponse> {
  const res = await fetch(`${BASE_URL}/api/accuracy/run-all`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(req),
  });
  return handleResponse<JobResponse>(res);
}

export async function runIncidents(req: RunIncidentsRequest): Promise<JobResponse> {
  const res = await fetch(`${BASE_URL}/api/accuracy/run-incidents`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(req),
  });
  return handleResponse<JobResponse>(res);
}

export async function discoverIncidents(req: DiscoveryRequest): Promise<DiscoveryResponse> {
  const res = await fetch(`${BASE_URL}/api/accuracy/discover`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(req),
  });
  return handleResponse<DiscoveryResponse>(res);
}

File: src/gui/api/accuracy.py
#!/usr/bin/env python3
"""
Accuracy Testing API
====================

Client methods for the accuracy testing endpoints.
"""

from typing import Any, Dict, List, Optional

from gui.api.client import ApiClient


def list_scripts(client: ApiClient) -> List[str]:
    """Fetch available accuracy validation script names."""
    return client.get("/api/accuracy/scripts")


def run_validation(client: ApiClient, payload: Dict[str, Any]) -> Dict[str, Any]:
    """Run a single validation script.

    Args:
        client: API client instance.
        payload: ``RunValidationRequest`` fields (camelCase keys).

    Returns:
        Job response dict with ``id``, ``status``, etc.
    """
    return client.post("/api/accuracy/run", payload)


def run_all_validations(client: ApiClient, payload: Dict[str, Any]) -> Dict[str, Any]:
    """Run all validation scripts in sequence.

    Args:
        client: API client instance.
        payload: ``RunAllRequest`` fields.

    Returns:
        Job response dict.
    """
    return client.post("/api/accuracy/run-all", payload)


def run_incidents(client: ApiClient, payload: Dict[str, Any]) -> Dict[str, Any]:
    """Run selected incident validations.

    Args:
        client: API client instance.
        payload: ``RunIncidentsRequest`` fields including ``incidents`` list.

    Returns:
        Job response dict.
    """
    return client.post("/api/accuracy/run-incidents", payload)


def discover_incidents(
    client: ApiClient, input_directory: str
) -> Dict[str, Any]:
    """Auto-discover incident files in a directory.

    Args:
        client: API client instance.
        input_directory: Path to scan for incident CSV files.

    Returns:
        Discovery response with ``results`` and ``totalFound``.
    """
    return client.post(
        "/api/accuracy/discover", {"inputDirectory": input_directory}
    )
